technical-reports/format/composite-types.md

```diff
@@ -60,155 +75,205 @@ Other examples might be:
+<div style="height: 2rem; background: linear-gradient(90deg, #ffff00 66.6%, #ff0000);"></div>
 
-For example, imagine Sketch wanted to export all layer styles from a Sketch document to a token file. The Sketch app could have a built-in type definition that corresponds to layer styles. Perhaps something like this:
+</aside>
 
-<aside class="example" title="Custom type definitions for Sketch">
+<aside class="example" title="Gradient token using references example">
 
 ```json
 {
-  "sketch layer style": {
-    "type": "typedef",
-    "value": {
-      "fill": {
-        "type": "color",
-        "required": false
+  "brand-primary": {
+    "type": "color",
+    "value": "#99ff66"
+  },
+
+  "position-end": {
+    "value": 1
+  },
+
+  "brand-in-the-middle": {
+    "type": "gradient",
+    "value": [
+      {
+        "color": "#000000",
+        "position": 0
       },
-      "border": {
-        "color": {
-          "type": "color",
-          "required": false
-        },
-        "width": {
-          "type": "dimension",
-          "required": false
-        }
+      {
+        "color": "{brand-primary}",
+        "position": 0.5
+      },
+      {
+        "color": "#000000",
+        "position": "{position-end}"
       }
-    }
+    ]
   }
 }
 ```
 
+Describes a color token called "brand-primary", which is referenced as the mid-point of a gradient is black at either end.
+
+<div style="height: 2rem; background: linear-gradient(90deg, #000000, #99ff66, #000000);"></div>
+
 </aside>
 
-It could then save out that type definition into the token file it exports. Other tools could then make use of that same type definition and do useful things with tokens of that type.
+## Typography
 
-### Custom configuration
+Represents a typographic style. The type property must be set to the string “typography”. The value must be an object with the following properties:
 
-Via configuration files or settings menus, tools could let users define custom behavior for types they have defined. For instance, in an export tool like Style Dictionary, users might be able to configure a custom transform or formatter for their composite types.
-Then, this example...
+- `fontName`: The typography's font. The value of this property must be a valid [font name](#font-name) or a reference to a font name token.
+- `fontSize`: The size of the typography. The value of this property must be a valid [dimension value](#dimension) or a reference to a dimension token.
+- `fontWeight`: The weight of the typography. The value of this property must be a valid JSON string or a reference to a string token.
+- `letterSpacing`: The horizontal spacing between characters. The value of this property must be a valid [dimension value](#dimension) or a reference to a dimension token.
+- `lineHeight`: The vertical spacing between lines of typography. The value of this property must be a valid JSON string or a reference to a string token.
```

Comment:
Wouldn't a number or dimension (or both) be more appropriate here?

Otherwise, if it's a string value, then what's the expected syntax within that string value?

Reply:
CSS often uses unitless line heights, which would push me toward "number" for the type, but that would exclude the definition of a line height that includes the unit, hence "string"

Our current dimension Type requires either `px` or `rem` excluding the possibility of using unitless line heights,

Reply:
Why not both? Could we not say the `lineHeight` sub-value must be a valid dimension _or_ number value?

Perhaps the ability for sub-types to be one of several types could be (yet) another special quality of composite types?

Reply:
`lineHeight` could be a type itself, which is a union type of: number, dimension, percentage, or keyword 'normal' (based on the CSS spec at least: https://developer.mozilla.org/en-US/docs/Web/CSS/line-height)

Reply:
I think we should allow users to define unitless line heights. Hence, I agree with @dbanksdesign and I think we could add a `lineHeight` type.